pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -196,8 +197,17 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
 		}
 	} else {
 		// bring up a new mon to replace the unhealthy mon
-		if err := c.failoverMon(name); err != nil {
-			logger.Errorf("failed to failover mon %q. %v", name, err)
+		// only possible if we have enough "ready nodes"
+		if readyNodes, err := k8sutil.GetKubernetesReadyNodes(c.context.Clientset); err != nil {
+			logger.Errorf("list of <Ready> nodes for the mon %q not available: %v", name, err)
+		} else {
+			if len(readyNodes) < desiredMonCount {
+				logger.Errorf("Not enough <Ready> nodes(only %d) to schedule the mon %q. It is needed at least %d nodes. Canceling failover", len(readyNodes), name, desiredMonCount)
```

Comment:
nit: "It is needed at least %d nodes" -> "At least %d nodes are needed"

Reply:
nit: you can `return` after this error, then you don't need the `else` on line 206

Reply:
Error messages should not start with capitals. Also, more detailed reasoning should be later in the description; I think the ordering could be flipped.
```suggestion
				logger.Errorf("caneling failover. not enough <Ready> nodes (%d ready, %d needed) to schedule mon %q", len(readyNodes), desiredMonCount, name)
```